mne/utils/progressbar.py

```diff
@@ -5,99 +5,72 @@
 # License: BSD (3-clause)
 
 from collections.abc import Iterable
-import time
-import logging
-import tempfile
-import sys
-import os.path as op
 import os
-import shutil
+import os.path as op
+import tempfile
+from threading import RLock
 
 import numpy as np
 
-from ._logging import logger
-from .misc import sizeof_fmt
+
+def _noop():
+    pass
 
 
 class ProgressBar(object):
     """Generate a command-line progressbar.
 
     Parameters
     ----------
-    max_value : int | iterable
-        Maximum value of process (e.g. number of samples to process, bytes to
-        download, etc.). If an iterable is given, then `max_value` will be set
-        to the length of this iterable.
+    iterable : iterable | None
+        ``max_value` will be set to the length of this iterable.
```

Comment:
```suggestion
        ``max_value`` will be set to the length of this iterable.
```

Reply:
besides the missing backtick, since `max_value` is itself a parameter, shouldn't this say "if max_value is None, it will be set to the length of this iterable"?